habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -469,6 +469,163 @@ def step(self, *args, is_last_action, **kwargs):
             return {}
 
 
+@registry.register_task_action
+class BaseVelSpotAction(ArticulatedAgentAction):
+    """
+    The articulated agent base motion is constrained to the NavMesh and controlled with velocity commands integrated with the VelocityControl interface.
+
+    Optionally cull states with active collisions if config parameter `allow_dyn_slide` is True
+    """
+
+    def __init__(self, *args, config, sim: RearrangeSim, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.base_vel_ctrl = habitat_sim.physics.VelocityControl()
+        self.base_vel_ctrl.controlling_lin_vel = True
+        self.base_vel_ctrl.lin_vel_is_local = True
+        self.base_vel_ctrl.controlling_ang_vel = True
+        self.base_vel_ctrl.ang_vel_is_local = True
+        self._allow_dyn_slide = self._config.get("allow_dyn_slide", True)
+        self._lin_speed = self._config.lin_speed
+        self._ang_speed = self._config.ang_speed
+        self._allow_back = self._config.allow_back
+        self._cylinder_deviate = self._config.cylinder_deviate
+        self._lin_collision_threshold = self._config.lin_collision_threshold
+        self._ang_collision_threshold = self._config.ang_collision_threshold
+
+    @property
+    def action_space(self):
```

Comment:
is this the same as action_space() from BaseVelAction?

Reply:
Yes, it is.